ocaml/typing/mtype.ml

```diff
@@ -716,7 +716,7 @@ let collect_arg_paths mty =
 
 type remove_alias_args =
     { mutable modified: bool;
-      exclude: Ident.t -> Path.t -> bool;
+      exclude: Ident.t option -> Path.t -> bool;
```

Comment:
The meaning of this option is pretty opaque.  Admittedly, `exclude` was already a little subtle, since one implementation only cares about the path and the other only about the name.  But I think it would be clearer to make a new type like:
```
type path_from =
| Alias of Ident.t
| Strengthening
```
(or, if that feels like overkill, to leave a comment here documenting the same).